### ISSUE TITLE
ipfs: consider ipld-prime and ipld-selectors

### ISSUE THREAD

**Issue by liamsi** [celestiaorg/celestia-node#6]
Another interesting suggestion brought up by @Wondertan is to use ipld-prime.   Particularly, the IPLD selectors look extremely cool and relevant to what we are doing. E.g. they could be used to download a range of leafs that belong to a particular namespace at once (#221). 

---

Note: If it turns out much simpler to use ipld-selectors instead of manually adding logic to traverse the NMT for celestiaorg/celestia-node#5, we should prioritize this issue. Let's keep in mind though that ipld-prime is a relatively young project and it might be better to wait until it matures and stabilizes a bit further.

---

- implementation: https://github.com/ipld/go-ipld-prime
- selectors: https://github.com/ipld/specs/blob/master/selectors/selectors.md
- selectors package: https://github.com/ipld/go-ipld-prime/tree/master/traversal/selector
- blog-post with examples: https://www.memoryandthought.me/go,/ipld/2021/01/23/go-ipld-prime-intro.html

**Comment by Wondertan** [comment]
Another reason why we should not rush with Graphsync is that it depends on IPLDv1 and it's currently under major refactoring. I would suggest not to touch this topic for several months and return to this only after devnet.

**Comment by liamsi** [comment]
closing in favour of #614
